src/main/java/frc/robot/subsystems/swerve/GyroIOPigeon2.java

```diff
@@ -53,9 +54,8 @@ public void updateInputs(GyroIOInputs inputs, List<Samples> asyncOdometrySamples
     inputs.odometryYawPositions =
         asyncOdometrySamples.stream()
             .map(s -> s.values().get(yaw))
-            .filter(s -> s != null)
-            .map(Rotation2d::fromDegrees)
-            .toArray(Rotation2d[]::new);
+            .map(d -> d == null ? new NullableRotation2d(null) : NullableRotation2d.fromDegrees(d))
```

Comment:
Minor usage improvement - you might consider adding a helper that does this. Think an analog to Optional.ofNullable. The only awkwardness is the degrees/rotations question. `ofNullableDegrees`?

Or not. Not a major change.

Reply:
Ah, I see you've already got a `fromDegrees` but it doesn't take a `Double` so it can't be null. You could pretty easily repurpose that to do this.